### ISSUE TITLE
cpanel migrate - a lot of garbage files

### ISSUE THREAD

**Issue by shoulders** [virtualmin/virtualmin-gpl#941]
| SYSTEM INFORMATION ||
|----------|----------|
|  OS type and version |  Ubuntu Linux 22.04.5             |
|  Usermin version     |  2.102                            |
|  Virtualmin version  |  7.20.2 Pro                       |
|  Theme version       |  21.20.7                          |
|  Apache version      |  2.4.52                           |
|  Package updates     |  9 package updates are available  |

## the issue

When you import a cpanel account it seems to bring a load of gargbage files into the home directory and some subfolders such as /home/etc/

@jcameron the cpanel account I have sent you should show you. I can compile a full list if required but this would take some time and might be a weekend job.

sorry to be a bit lazy on this one.

**Comment by iliajie** [comment]
Hello,

Are you sure that migrating a cPanel backup actually creates a `/home/etc` directory? That seems bizarre, and if it does, it’s definitely a bug.

**Comment by shoulders** [comment]
Let me investigate more at the weekend and I will compile a list of garbage files and folders that are imported or prove myself wrong 😃

But if you import the archive I sent Jamie that should show you.

**Comment by jcameron** [comment]
Do you mean `/home/etc` or `/home/migrateduser/etc` ? The latter is expected, as all new Virtualmin domains should have it.

**Comment by shoulders** [comment]
> Do you mean /home/etc or /home/migrateduser/etc ? The latter is expected, as all new Virtualmin domains should have it.

No I mean all over the place in the users home folder `/home/myuser/` 

It is almost like you are just expanding the cpanel archive directly into the virtualmin user folder for some things. I need to fully go over the users folder to properly say what is garbage and what is not.

I will need to manually clean my virtualmin user's folder, this is going to take me ages and I don't even know where to start so this needs to be looked at for people in the future etc... 😄 

Just a quick one is: `/home/myuser/.spamassassin/` and there are loads of files from my old server. See below. The name was my old servers name.

![image](https://github.com/user-attachments/assets/c4cc34a2-6609-497e-ad8d-ed232f068587)

**Comment by jcameron** [comment]
Yes we currently copy over everything from the `homedir` directory in the backup, or `homedir.tar` if it exists. If there are directories you think should be excluded, please let me know!

**Comment by shoulders** [comment]
Is there a reason for just copying everything over?

 

Is the virtualmin folder structure the same as cpanel?

**Comment by iliajie** [comment]
We copy all files because users may have created their own custom directories.

**Comment by shoulders** [comment]
That makes sense.

Let me check over the import to see what should not be moved.

**Comment by iliaross** [comment]
Reopened, as this was closed by an automated cleanup run. @shoulders, sorry for the noise, and thanks for the contribution. We definitely didn't mean any disrespect.

If it turns out the issue is already resolved for you, you're welcome to close it.
